esmvaltool/utils/recipe_filler.py

```diff
@@ -0,0 +1,684 @@
+            if "exp" in var_pars:
+                if isinstance(var_pars["exp"],
+                              list) and "ensemble" not in var_pars:
+                    logger.error("Asking for experiments list for ")
+                    logger.error(f"variable {var_name} - you need to ")
+                    logger.error("define an ensemble for this case.")
+                    do_exit = True
+    if do_exit:
+        logger.error("Please fix the issues in recipe and rerun. Exiting.")
+        sys.exit(1)
+
+
+def _check_config_file(user_config_file):
+    """Perform a quick recipe check for mandatory fields."""
+    do_exit = False
+    if "rootpath" not in user_config_file:
+        logger.error("Config file missing rootpath section.")
+        do_exit = True
+    if "drs" not in user_config_file:
+        logger.error("Config file missing drs section.")
+        do_exit = True
+    for proj in cmip_eras:
+        if proj not in user_config_file["rootpath"].keys():
+            logger.error(f"Config file missing rootpath for {proj}")
+            do_exit = True
+        if proj not in user_config_file["drs"].keys():
+            logger.error(f"Config file missing drs for {proj}")
+            do_exit = True
+    if do_exit:
+        logger.error("Please fix issues in config file and rerun. Exiting.")
+        sys.exit(1)
+
+
+def _parse_recipe_to_dicts(yamlrecipe):
+    """Parse a recipe's variables into a dictionary of dictionairies."""
+    output_dicts = {}
+    for diag in yamlrecipe['diagnostics']:
+        for variable, var_dict in yamlrecipe['diagnostics'][diag][
+                'variables'].items():
+            new_dict = base_dict.copy()
+            for var_key, var_value in var_dict.items():
+                if var_key in new_dict:
+                    new_dict[var_key] = var_value
+            output_dicts[(diag, variable)] = new_dict
+
+    return output_dicts
+
+
+def _add_datasets_into_recipe(additional_datasets, output_recipe):
+    """Add the datasets into a new recipe."""
+    yaml = YAML()
+    yaml.default_flow_style = False
+    with open(output_recipe, 'r') as yamlfile:
+        cur_yaml = yaml.load(yamlfile)
+        for diag_var, add_dat in additional_datasets.items():
+            if add_dat:
+                if 'additional_datasets' in cur_yaml['diagnostics']:
+                    cur_yaml['diagnostics'][diag_var[0]]['variables'][
+                        diag_var[1]]['additional_datasets'].extend(add_dat)
+                else:
+                    cur_yaml['diagnostics'][diag_var[0]]['variables'][
+                        diag_var[1]]['additional_datasets'] = add_dat
+    if cur_yaml:
+        with open(output_recipe, 'w') as yamlfile:
+            yaml.dump(cur_yaml, yamlfile)
+
+
+def _find_all_datasets(recipe_dict, cmip_eras):
+    """Find all datasets explicitly."""
+    datasets = []
+    for cmip_era in cmip_eras:
+        if cmip_era == "CMIP6":
+            activity = "CMIP"
```

Comment:
Do we really want to hardcode this? It is possible to find scenario data somehow (with activity `ScenarioMIP`)? I think there should be a function in ESMValCore that searches for the correct activity.

Reply:
yeah! good point, but for that I'd have to plug moar functionality including the OBS stuff, which is on the to-do list, I'll do it then, let's just keep this a bit basic for now, the PR is big enough :+1: